frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -64,20 +65,90 @@ local function tidy_markup(results)
     return results
 end
 
+function ReaderDictionary:cleanSelection(text, is_single_word)
+    -- Will be used by ReaderWikipedia too (with is_single_word false)
+    -- We do multiple times the same replacements, which is most of the time overkill,
+    -- but sometimes provices better cleaning
```

Comment:
I removed quite a bit of these gsub. Tested on some extrem cases, and it works as fine.
I don't think we have to worry much about time spend in these gsubs. Compared to the time it will take to do the stardict or wikipedia lookup, it's nothing. And we do that only when the user asks for a lookup.
Anyway, i think it looks cleaner, and clear alongside my samples in comments.